bundles/org.openhab.core/src/main/java/org/openhab/core/types/util/UnitUtils.java

```diff
@@ -112,6 +112,17 @@ public class UnitUtils {
      * @return The Dimension string or null if the unit can not be found in any of the SystemOfUnits.
      */
     public static @Nullable String getDimensionName(Unit<?> unit) {
+        String compatibleDimension = getDirectDimensionName(unit);
+        if (compatibleDimension == null) {
+            compatibleDimension = getDimensionName(unit.inverse());
```

Comment:
Does unit.inverse() always return a non null value?

Reply:
Yes. It may not make immediate logical sense (what's Meter^-1?), but it always has a value.